table/tables/mutation_checker.go

```diff
@@ -136,7 +136,8 @@ func checkHandleConsistency(rowInsertion mutation, indexMutations []mutation, in
 		if err != nil {
 			return errors.Trace(err)
 		}
-		if indexHandle.Compare(insertionHandle) != 0 {
+		// NOTE: handle type can be different, see issue 29520
+		if indexHandle.IsInt() == insertionHandle.IsInt() && indexHandle.Compare(insertionHandle) != 0 {
```

Comment:
How about the common handle type?

Reply:
`indexHandle.IsInt() == insertionHandle.IsInt()`
will evaluate to 
`0 == 0`

Reply:
Do we need to compare and check if `indexHandle.IsCommon() == insertionHandle.IsCommon()" ?

Reply:
There's no such a method. I think we are only using `IsInt()` to distinguish handle types.

Reply:
I just notice there is an `Equal` method in the Handle interface which checks the type:
```go
func (ih IntHandle) Equal(h Handle) bool {
	return h.IsInt() && int64(ih) == h.IntValue()
}
```
@ekexium  Can we use this instead?

Reply:
Then the problem is what we should do if `Equal` returns false. We can't tell if it's type mismatch or not.

Reply:
Since we have only two kind of handles that are usable now, I think `indexHandle.IsInt() == insertionHandle.IsInt()` is enough.
Maybe a better way is to add a method to get the kind, and check something like `indexHandle.Kind() == insertionHandle.Kind()`.

Reply:
Seems we check the handle content if it's int type, if it's clustered index handle type or common type do we need to check if the content does match?

Reply:
@cfzjywxk The expression `indexHandle.IsInt() == insertionHandle.IsInt()` checks equality of two bools, rather than `AND`. Both `false`s evaluates to true.

Reply:
@cfzjywxk IMO it's better to check it if we can.

Reply:
This adds some implicit dependencies that there are two types of handle, as this constraint will not change very much seems it's fine.